app/code/Magento/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php

```diff
@@ -92,4 +94,38 @@ private function getSerializer()
         }
         return $this->serializer;
     }
+
+    /**
+     * Add Value Sort To Collection Select
+     * all NULL values will add to the end
+     *
+     * @param \Magento\Eav\Model\Entity\Collection\AbstractCollection $collection
+     * @param string $dir direction
+     * @return \Magento\Eav\Model\Entity\Attribute\Source\AbstractSource
+     */
+    public function addValueSortToCollection($collection, $dir = Collection::SORT_ORDER_DESC) : AbstractSource
+    {
+        $attributeCode = $this->getAttribute()->getAttributeCode();
+        $attributeId = $this->getAttribute()->getId();
+        $attributeTable = $this->getAttribute()->getBackend()->getTable();
+        $linkField = $this->getAttribute()->getEntity()->getLinkField();
+
+        $tableName = $attributeCode . '_t';
+
+        $collection->getSelect()->joinLeft(
```

Comment:
Attribute can has values in scope of different stores.
Test scenario:
1. Create product.
2. Open edit form and set 'Country of Manufacture' to 'USA' in Global Scope.
3. Save the product.
4. Open edit form in Store Scope and set ' Country of Manufacture' to 'China'.
5. Open product grid
6. Try to sort by ' Country of Manufacture'

E.R.
1. Sorted products by the column value.

A.R. 
1. Error message.

Please check this case.

Reply:
fixed

Reply:
Hi @XxXgeoXxX,

Please check next cases:
1. Attribute can be switched to 'Global' scope.
2. For now, sort order will works only for value in default scope.